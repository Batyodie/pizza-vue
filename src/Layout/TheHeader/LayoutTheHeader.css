@import "../../style/Base/Extend/BaseExtend.css";

.Wrapper {
  @extend .ContentWrapper;
}
.BorderBottom {
  position: absolute;
  bottom: 0;
  max-width: 1340px;
  width: 100%;
  z-index: 10;
  left: 0;
  height: 1px;
  border-bottom: 1px solid #f7f7f7;
}

@media (max-width: 690px) {
  .Wrapper {
    justify-content: flex-start;
  }
  .LogoMargin {
    margin-top: 1rem;
    margin-bottom: 1rem;
  }
  .ButtonMargin {
    @extend .LogoMargin;
    margin-top: 1.1rem;
  }
}

@media (max-width: 374px) {
  .Wrapper {
    padding-left: 0px;
    padding-right: 0px;
    justify-content: center;
  }
}
