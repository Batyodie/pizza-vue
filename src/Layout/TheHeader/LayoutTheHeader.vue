<script>
import { Button, Logo } from "@/components";
import { BtnStyle } from "@/components/style";
import { mapGetters } from "vuex";

import { iconfinder } from "@/assets";

import { LayoutTheHeader } from "@/Layout/style";
export default {
  name: "LayoutTheHeader",
  components: { Logo, Button },
  // The flag for the button, which is responsible for the appearance of the button in the header layout
  props: {
    HeaderButton: {
      type: Boolean,
      required: true
    },
    // sub-title test for component
    sloganText: {
      type: String,
      required: true,
      default: ""
    }
  },

  data() {
    return {
      icon: iconfinder,
      IsActiveButton: this.HeaderButton,
      LogoTitle: this.sloganText
    };
  },

  computed: {
    ...mapGetters({
      totalPrice: "getTotalPrice",
      itemsCount: "getItemsCount"
    }),

    BtnStyle() {
      return BtnStyle;
    },

    LayoutTheHeader() {
      return LayoutTheHeader;
    }
  }
};
</script>

<template>
  <header :class="LayoutTheHeader.Wrapper">
    <div :class="LayoutTheHeader.BorderBottom"></div>
    <Logo :sloganText="sloganText" :class="LayoutTheHeader.LogoMargin" />
    <template v-if="IsActiveButton">
      <router-link name="basket" to="/basket" style="text-decoration: none">
        <Button :class="[BtnStyle.Basket, LayoutTheHeader.ButtonMargin]">
          <template slot="Icon">
            <img
              style="margin: 1px 0 0 13px;"
              :src="icon"
              width="15"
              height="15"
              alt="icon basket"
            />
            <span :class="BtnStyle.Body">{{ itemsCount }}</span>
          </template>
          <template slot="IconBorder">
            <span :class="BtnStyle.IconBorder"></span>
          </template>
          <template slot="ButtonText"> {{ totalPrice }}</template>
        </Button>
      </router-link>
    </template>
  </header>
</template>
