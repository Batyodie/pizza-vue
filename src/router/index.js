import Vue from "vue";
import VueRouter from "vue-router";
import Home from "../views/Market";
import Basket from "../views/Basket/Basket";

Vue.use(VueRouter);

const routes = [
  {
    path: "/",
    name: "Home",
    component: Home,
  },
  {
    path: "/basket",
    name: "Basket",
    component: Basket,
  },
];

const router = new VueRouter({
  mode: "history",
  base: process.env.BASE_URL,
  hash: false,
  routes,
});

export default router;
