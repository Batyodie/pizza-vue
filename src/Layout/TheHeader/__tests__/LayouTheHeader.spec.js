import { createLocalVue, mount } from "@vue/test-utils";
// dropdwons componens files
import TheHeader from "@/Layout/TheHeader/LayoutTheHeader";
import VueRouter from "vue-router";
import router from "@/router";

const localVue = createLocalVue();
localVue.use(VueRouter);

// fake props
const sloganText = "Some fake title";
const HeaderButton = true;

const itemsCount = 0;
const totalPrice = 0;

describe("Unit tests for LayoutTheHeader component", () => {
  const wrapper = mount(TheHeader, {
    localVue,
    router,
    propsData: {
      sloganText,
      HeaderButton,
    },
    computed: {
      totalPrice() {
        return totalPrice;
      },
      itemsCount() {
        return itemsCount;
      },
    },
  });

  it("should initialize correctly", () => {
    // correctly Vue instance
    expect(wrapper.exists()).toBeTruthy();
    // We expect the wrapper to be the TheHeader component
    expect(wrapper.is(TheHeader)).toBeTruthy();
    //  We expect the wrapper props to equal the fake props
    expect(wrapper.vm.sloganText).toEqual(sloganText);
    expect(wrapper.vm.HeaderButton).toEqual(HeaderButton);
  });

  it("should render correctly logo title", () => {
    const renderSlogan = `${sloganText}`;

    // Expect the wrapper to be the slogan text render
    expect(wrapper.text(renderSlogan)).toBeTruthy();
    // Expect thatrender SloganText in Slogan selector equal to renderSlogan
    expect(wrapper.find(".Slogan > p").text()).toEqual(renderSlogan);
  });

  it("should render correctly basket button", () => {
    const renderCount = `${"0"}   ${"0"}`;
    const btn = wrapper.find(".Basket");
    const bodyBtnAll = btn.findAll(".Body");
    // Expect the wrapper to be the button text render
    expect(btn.text(renderCount)).toBeTruthy();
    // Expect the wrapper to be the button text render to equal render count
    expect(btn.text()).toEqual(renderCount);
    expect(bodyBtnAll.at(0).text()).toEqual("0");
    expect(bodyBtnAll.at(1).text()).toEqual("0");
  });

  it("Should match the snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
