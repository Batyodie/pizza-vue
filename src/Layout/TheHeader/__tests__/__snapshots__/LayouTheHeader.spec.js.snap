// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for LayoutTheHeader component Should match the snapshot 1`] = `
<header class="Wrapper">
  <div class="BorderBottom"></div> <a href="/" aria-current="page" class="router-link-exact-active router-link-active LogoMargin" style="text-decoration: none;">
    <div class="Inner"><img src="" width="38" height="38" alt="Main logo" class="Image">
      <div class="Slogan">
        <h1 class="SloganTitle">VUE PIZZA</h1>
        <p class="SloganText">Some fake title</p>
      </div>
    </div>
  </a> <a href="/basket" class="" name="basket" style="text-decoration: none;"><button type="button" class="Btn Basket ButtonMargin"><span class="Body"> 0</span> <span class="IconBorder"></span> <img src="" width="15" height="15" alt="icon basket" style="margin: 1px 0px 0px 13px;"> <span class="Body">0</span></button></a>
</header>
`;
